Sources/Subjects/ReplaySubject.swift

```diff
@@ -0,0 +1,171 @@
+//
+//  ReplaySubject.swift
+//  CombineExt
+//
+//  Created by Jasdev Singh on 13/04/2020.
+//  Copyright © 2020 Combine Community. All rights reserved.
+//
+
+import Combine
+
+/// `ReplaySubject` is a [CurrentValueSubject](https://developer.apple.com/documentation/combine/currentvaluesubject)
+/// that can buffer, well, more than one value! It stores value events, up to its `maxBufferSize` in a
+/// [first-in-first-out](https://en.wikipedia.org/wiki/FIFO_(computing_and_electronics)) manner and then replays it to
+/// future subscribers and also forwards completion events.
+///
+/// The implementation borrows heavily from [Tristan’s](https://github.com/tcldr/Entwine/blob/b839c9fcc7466878d6a823677ce608da998b95b9/Sources/Entwine/Operators/ReplaySubject.swift)
+/// with the main modifications being leaning on Shai’s [DemandBuffer](https://github.com/CombineCommunity/CombineExt/blob/7ed4677e33fdc963af404423cb563e133c271d2b/Sources/Common/DemandBuffer.swift)
+/// instead of `Entwine`’s [SinkQueue](https://github.com/tcldr/Entwine/blob/b839c9fcc7466878d6a823677ce608da998b95b9/Sources/Common/Utilities/SinkQueue.swift)
+/// and a plain ol’ `[Output]` buffer instead of a
+/// [custom, LinkedListQueue-backed one](https://github.com/tcldr/Entwine/blob/8be24a59bc91410bb29e84b1c4ae35398a5839c8/Sources/Entwine/Operators/ReplaySubject.swift#L162-L177).
+public final class ReplaySubject<Output, Failure: Error> {
+    // MARK: - Private
+
+    private let maxBufferSize: UInt
+    private var buffer = [Output]()
+
+    // MARK: - Internal
+
+    var subscriptions = [Subscription<AnySubscriber<Output, Failure>>]()
+
+    // We also track subscriber identifiers, to more quickly bottom-out double subscribes instead of having to do a
+    // linear pass over `subscriptions`.
+    var subscriberIdentifiers = Set<CombineIdentifier>()
+
+    private var completion: Subscribers.Completion<Failure>?
+    private var isActive: Bool { completion == nil }
+
+    /// The initialization point for `ReplaySubject`s.
+    /// - Parameter maxBufferSize: The maximum number of value events to buffer and replay to all future subscribers.
+    public init(maxBufferSize: UInt) {
```

Comment:
same thing with the UInt, don't see a value in this (would use Int across the board for the sake of ergonomics)